### PR TITLE
replace int32_t with index_t in gpl

### PR DESCRIPTION
**Pull request by MikeLing**
Sorry, I thought I make this pr yesterday but actually I'm not :/

### CODE REVIEW

**Comment by karlnapf** [comment]
@vigsterkr so this is like `data` now ...  we merge things in here and then @MikeLing does update the `shogun-gpl` submodule hash in the main repo and then we get tests
Feel free to merge

File: shogun/classifier/FeatureBlockLogisticRegression.cpp [hunk 1]
```diff
@@ -250,14 +250,14 @@ SGVector<float64_t> CFeatureBlockLogisticRegression::apply_get_outputs(CFeatures
...
-	int32_t num=features->get_num_vectors();
+	auto num=features->get_num_vectors();
 	SGVector<float64_t> w = get_w();
 	ASSERT(num>0)
 	ASSERT(w.vlen==features->get_dim_feature_space())
 
 	float64_t* out=SG_MALLOC(float64_t, num);
 	features->dense_dot_range(out, 0, num, NULL, w.vector, w.vlen, bias);
-	for (int32_t i=0; i<num; i++)
+	for (index_t i=0; i<num; i++)
```
Comment (vigsterkr) [inline shogun/classifier/FeatureBlockLogisticRegression.cpp]: use `auto` here.

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 2]
```diff
@@ -143,16 +143,16 @@ SGVector<float64_t> CWDSVMOcas::apply_get_outputs(CFeatures* data)
...
 	SG_FREE(wd_weights);
 	wd_weights=SG_MALLOC(float32_t, degree);
 	SG_FREE(w_offsets);
-	w_offsets=SG_MALLOC(int32_t, degree);
-	int32_t w_dim_single_c=0;
+	w_offsets=SG_MALLOC(index_t, degree);
+	index_t w_dim_single_c=0;
 
-	for (int32_t i=0; i<degree; i++)
+	for (index_t i=0; i<degree; i++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: is the type of degree `index_t`? as it'd be better to use `auto i` but for that degree needs to be index_t

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 3]
```diff
@@ -322,15 +322,15 @@ void* CWDSVMOcas::add_new_cut_helper( void* ptr)
 	//float32_t* new_a = SG_MALLOC(float32_t, nDim);
 	//memset(new_a, 0, sizeof(float32_t)*nDim);
 
-	int32_t* val=SG_MALLOC(int32_t, cut_length);
-	for (int32_t j=start; j<end; j++)
+	index_t* val=SG_MALLOC(index_t, cut_length);
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: `auto`

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 4]
```diff
@@ -322,15 +322,15 @@ void* CWDSVMOcas::add_new_cut_helper( void* ptr)
...
+	for (index_t j=start; j<end; j++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: `auto`? is `start` and `end` `index_t`?

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 5]
```diff
@@ -322,15 +322,15 @@ void* CWDSVMOcas::add_new_cut_helper( void* ptr)
...
-		memset(val,0,sizeof(int32_t)*cut_length);
-		int32_t lim=CMath::min(degree, string_length-j);
-		int32_t len;
+		index_t offs=o->w_dim_single_char*j;
+		memset(val,0,sizeof(index_t)*cut_length);
+		auto lim=CMath::min(degree, string_length-j);
+		index_t len;
 
-		for (int32_t k=0; k<lim; k++)
+		for (index_t k=0; k<lim; k++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: `auto`? (if lim is index_t)

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 6]
```diff
@@ -461,33 +461,33 @@ void* CWDSVMOcas::compute_output_helper(void* ptr)
...
+	index_t* w_offsets = o->w_offsets;
 	float32_t* wd_weights = o->wd_weights;
 	float32_t* w= o->w;
 
 	float64_t* y = o->lab;
 	float64_t normalization_const = o->normalization_const;
 
 
-	for (int32_t j=0; j<string_length; j++)
+	for (index_t j=0; j<string_length; j++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: `auto`? if string_length is index_t

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 7]
```diff
@@ -461,33 +461,33 @@ void* CWDSVMOcas::compute_output_helper(void* ptr)
...
 	{
-		int32_t offs=o->w_dim_single_char*j;
-		for (int32_t i=start ; i<end; i++)
+		index_t offs=o->w_dim_single_char*j;
+		for (index_t i=start ; i<end; i++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: auto?

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 8]
```diff
@@ -533,7 +533,7 @@ void* CWDSVMOcas::compute_output_helper(void* ptr)
 		}
 	}
 
-	for (int32_t i=start; i<end; i++)
+	for (index_t i=start; i<end; i++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: auto?

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 9]
```diff
@@ -461,33 +461,33 @@ void* CWDSVMOcas::compute_output_helper(void* ptr)
...
 
-		for (int32_t k=0; k<lim; k++)
+		for (index_t k=0; k<lim; k++)
 		{
 			bool free_vec;
 			uint8_t* vec=f->get_feature_vector(j+k, len, free_vec);
 			float32_t wd = wd_weights[k];
 
-			for (int32_t i=start; i<end; i++) // quite fast 1.9s
+			for (index_t i=start; i<end; i++) // quite fast 1.9s
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: auto?

File: shogun/classifier/svm/WDSVMOcas.cpp [hunk 10]
```diff
@@ -461,33 +461,33 @@ void* CWDSVMOcas::compute_output_helper(void* ptr)
...
+		for (index_t i=start ; i<end; i++)
 			val[i]=0;
 
-		int32_t lim=CMath::min(degree, string_length-j);
-		int32_t len;
+		auto lim=CMath::min(degree, string_length-j);
+		index_t len;
 
-		for (int32_t k=0; k<lim; k++)
+		for (index_t k=0; k<lim; k++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.cpp]: auto?

File: shogun/classifier/svm/WDSVMOcas.h [hunk 11]
```diff
@@ -176,24 +176,24 @@ class CWDSVMOcas : public CMachine
...
-			int32_t len=0;
+			index_t len=0;
 			float64_t sum=0;
 			bool free_vec;
 			uint8_t* vec=features->get_feature_vector(num, len, free_vec);
 			//SG_INFO("len %d, string_length %d\n", len, string_length)
 			ASSERT(len==string_length)
 
-			for (int32_t j=0; j<string_length; j++)
+			for (index_t j=0; j<string_length; j++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.h]: auto?

File: shogun/classifier/svm/WDSVMOcas.h [hunk 12]
```diff
@@ -209,7 +209,7 @@ class CWDSVMOcas : public CMachine
 		{
 			ASSERT(features)
 			normalization_const=0;
-			for (int32_t i=0; i<degree; i++)
+			for (index_t i=0; i<degree; i++)
```
Comment (vigsterkr) [inline shogun/classifier/svm/WDSVMOcas.h]: auto?

**Comment by MikeLing** [comment]
ask review for this pr @vigsterkr :)

**Pull request merged**